src/event.rs

```diff
@@ -277,9 +298,17 @@ where
                 }
             }
         }
-        state.update(ctx)?;
-        state.draw(ctx)?;
-    }
 
-    Ok(())
+        if let Err(e) = state.update(ctx) {
```

Comment:
I think I should have moved this part into one of the event handlers, maybe MainEventsCleared?